qiskit_machine_learning/algorithms/kernel_trainers/quantum_kernel_trainer.py

```diff
@@ -0,0 +1,171 @@
+            feature_map=...,
+            free_parameters=...,
+            quantum_instance=...
+        )
+
+        qk_trainer = QuantumKernelTrainer(quant_kernel)
+
+        qsvc = QSVC(quantum_kernel=qk_trainer)
+        qsvc.fit(X_train, y_train)
+        score = qsvc.score(X_test, y_test)
+    """
+
+    def __init__(
+        self,
+        quantum_kernel: QuantumKernel,
+        loss: Optional[Union[str, KernelLoss]] = "svc_alignment",
+        optimizer: Optional[Optimizer] = None,
+        initial_point: Optional[np.ndarray] = None,
+    ):
+        """
+        Args:
+            quantum_kernel: A QuantumKernel object to be tuned
+            loss: A target loss function to be used in training. Default is `svc_alignment`,
+            optimizer: An instance of an optimizer to be used in training. When `None` defaults to SPSA.
+            initial_point: Initial point for the optimizer to start from.
+
+        Raises:
+            ValueError: unknown loss function
+        """
+        # Class fields
+        self._quantum_kernel = None
+        self._loss = None
+        self._optimizer = None
+        self._initial_point = None
+
+        # Setters
+        self.quantum_kernel = quantum_kernel
+        self.loss = loss
+        self.optimizer = optimizer
+        self.initial_point = initial_point
+
+    @property
+    def quantum_kernel(self):
+        """Returns the underlying quantum kernel."""
+        return self._quantum_kernel
+
+    @quantum_kernel.setter
+    def quantum_kernel(self, quantum_kernel: QuantumKernel) -> None:
+        """Sets the quantum kernel"""
+        self._quantum_kernel = quantum_kernel
+
+    @property
+    def loss(self):
+        """Returns the underlying loss function."""
+        return self._loss
+
+    @loss.setter
+    def loss(self, loss: Union[str, KernelLoss]) -> None:
+        """Sets the loss."""
+        if isinstance(loss, str):
+            loss = loss.lower()
+            if loss == "svc_alignment":
+                self._loss = SVCAlignment()
+            else:
+                raise ValueError(f"Unknown loss {loss}!")
+        elif callable(loss):
+            self._loss = loss
+        else:
+            raise ValueError(f"Unknown loss {loss}!")
+
+    @property
+    def optimizer(self) -> Optimizer:
+        """Returns an optimizer to be used in training."""
+        return self._optimizer
+
+    @optimizer.setter
+    def optimizer(self, optimizer: Optimizer) -> None:
+        """Sets the loss."""
+        if optimizer is None:
+            self._optimizer = SPSA(maxiter=50)
+        else:
+            self._optimizer = optimizer
+
+    @property
+    def initial_point(self) -> np.ndarray:
+        """Returns current initial point"""
+        return self._initial_point
+
+    @initial_point.setter
+    def initial_point(self, initial_point: np.ndarray) -> None:
+        """Sets the initial point"""
+        self._initial_point = initial_point
+
+    def fit_kernel(
```

Comment:
Would it make sense to take out the kernel from the trainer such that you pass it in on this method to fit it. I am wondering whether different trainers could be done such that QSVC was passed a kernel and a trainer as different instances so that is training was needed it could call the trainer with the kernel that was provided. This means you always pass a kernel and potentially an optional trainer.